### PR TITLE
MAINT: Added stacklevel argument to warnings.warn

### LINKED ISSUE #5945: Warning message for one function is used by several, and contains extraneous information
In [this Stackoverflow question](http://stackoverflow.com/questions/30679192/how-to-remove-multiple-values-from-an-array-at-once/) the inquirer points out the following discrepancy in an error message:

    In [5]: np.delete(x, x == a)
    /usr/lib/python2.7/dist-packages/numpy/lib/function_base.py:3254: FutureWarning: in the future insert will treat boolean arrays and array-likes as boolean index instead of casting it to integer
      "of casting it to integer", FutureWarning)

Since the warning is produced by a call to `delete()` it would be better if the message didn't refer to a specific function.

Also, why does the string ` "of casting it to integer"` appear at the end of the warning message?

### PR DESCRIPTION
**Pull request by sumitbinnani**
Updated `numpy/lib/function_base.py`.
Fixes  #5945

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1004,7 +1004,7 @@ def select(condlist, choicelist, default=0):
         # 2014-02-24, 1.9
         warnings.warn("select with an empty condition list is not possible"
                       "and will be deprecated",
-                      DeprecationWarning)
+                      DeprecationWarning, stacklevel = 2)
```
Comment (jaimefrio) [inline numpy/lib/function_base.py]: No spaces around the `=` in keyword assignments.

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -4206,7 +4209,8 @@ def insert(arr, obj, values, axis=None):
         # 2013-09-24, 1.9
         warnings.warn(
             "in the future the special handling of scalars will be removed "
-            "from insert and raise an error", DeprecationWarning)
+            "from insert and raise an error", DeprecationWarning,
+			stacklevel = 2)
```
Comment (jaimefrio) [inline numpy/lib/function_base.py]: I think this should be aligned with the beginning of the string, not with the opening parenthesis.

**Comment by jaimefrio** [comment]
Some style nitpicks...

**Comment by charris** [comment]
Umm, we don't allow tabs, use spaces please.

**Comment by sumitbinnani** [comment]
@charris  @jaimefrio 
Have fixed indentation issues and used spaces as advised. Kindly review the code and let me know if any more changes are required.

**Comment by sumitbinnani** [comment]
Hi @charris @jaimefrio 

Does the changes meet the requirements? Kindly let me know if any additional changes are required.

**Comment by sumitbinnani** [comment]
Hi @charris and @jaimefrio,

Is there any issue with the patch? Kindly let me know and would try to submit the patch for the same asap.

**Comment by njsmith** [comment]
The changes in the PR look good to me now.

Question 1: the PR summary says that this fixes #5945, but the main question in #5945 is about why a call to `delete` was issuing a warning that referred to `insert` -- does this PR actually address that? (Should it?)

Question 2: any chance you could add some tests for this? I guess adding tests for all the little deprecation warning tweaks would be tiresome and of limited value, because the code will just get removed again in a few versions anyway. But for the `RuntimeWarning`s in `cov`, `_median`, and `_percentile` it seems particularly useful, because the `RuntimeWarning`s will be around indefinitely, and because it's not 100% obvious whether stacklevel=2 is even correct for `_median` and `_percentile` which are internal helper functions. (It might be because I think the code the callers are written in C and thus don't affect the stacklevel= count, but it's not 100% obvious.) The way to do this would be to use the [`catch_warnings`](https://docs.python.org/2/library/warnings.html#testing-warnings) context manager with `record=True`, call the function that should issue a warning, and then check that the `.filename` attribute of the recorded warning object refers to the file that called the function rather than some internal numpy file.

Sorry for the slow response, and thanks for being persistent in pinging us!

**Comment by sumitbinnani** [comment]
Thanks @njsmith for the inputs. I will look into the details during weekend and update the thread.

**Comment by sumitbinnani** [comment]
Can you elaborate on what needs to be done, @homu?

**Comment by rkern** [comment]
@homu is a bot and will not reply. You will need to manually merge in `master` and fix the conflicts.

**Comment by charris** [comment]
This has been taken care of in #7148 along with a number of other cases, so closing this. Thanks @sumitbinnani, I apologize that we didn't get this in earlier.

**Pull request closed without merging**